Assets/Mirror/Runtime/NetworkAuthenticator.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Mirror
+{
+    /// <summary>
+    /// Unity Event for the NetworkConnection
+    /// </summary>
+    [Serializable] public class UnityEventConnection : UnityEvent<NetworkConnection> { }
+
+    /// <summary>
+    /// Base class for implementing component-based authentication during the Connect phase
+    /// </summary>
+    [AddComponentMenu("Network/NetworkAuthenticator")]
+    [HelpURL("https://mirror-networking.com/xmldocs/articles/Concepts/Authentication.html")]
```

Comment:
add Requirecomponent(typeof(NetworkManager))]
and [DontAllowMultipleComponents] please

Reply:
Please don't.    I would like to be able to distribute a prefab with my addons with the authenticator and UI.  Adding a [RequirComponent] forces the authenticator to be in the same gameobject as the networkmanager.